cmd/common.go

```diff
@@ -128,17 +133,34 @@ func setS3OptionsFromCLI(c *cli.Context) *v3.S3BackupConfig {
 	return s3BackupBackend
 }
 
-func doUpgradeLegacyCluster(ctx context.Context, kubeCluster *cluster.Cluster, fullState *cluster.FullState, flags cluster.ExternalFlags) error {
-	if _, err := os.Stat(kubeCluster.LocalKubeConfigPath); os.IsNotExist(err) {
-		// there is no kubeconfig. This is a new cluster
-		logrus.Debug("[state] local kubeconfig not found, this is a new cluster")
-		return nil
+func checkLegacyCluster(ctx context.Context, kubeCluster *cluster.Cluster, fullState *cluster.FullState, flags cluster.ExternalFlags) error {
+	stateFileExists, err := util.IsFileExists(kubeCluster.StateFilePath)
+	if err != nil {
+		return err
 	}
-	if _, err := os.Stat(kubeCluster.StateFilePath); err == nil {
-		// this cluster has a previous state, I don't need to upgrade!
+	if stateFileExists {
 		logrus.Debug("[state] previous state found, this is not a legacy cluster")
 		return nil
 	}
+	logrus.Debug("[state] previous state not found, possible legacy cluster")
+	return upgradePossibleLegacyCluster(ctx, kubeCluster, fullState, flags)
+}
+
+func upgradePossibleLegacyCluster(ctx context.Context, kubeCluster *cluster.Cluster, fullState *cluster.FullState, flags cluster.ExternalFlags) error {
```

Comment:
can we change the function name to reflect it's purpose better? as it's not going to do the upgrade